### PR TITLE
[RT001-584] Add webhook issuer filtering

### PR DESCRIPTION
**Pull request by mstrumeck**

### CODE REVIEW

File: src/saleor_app/deps.py [hunk 1]
```diff
@@ -121,6 +121,16 @@ async def verify_webhook_signature(
         )
 
 
+async def verify_issuing_principal_type(request: Request):
```
Comment (pkucmus) [inline src/saleor_app/deps.py:124]: This should or could be a class-based dependency:
```python
from saleor_app.schemas.webhook import Webhook


class IgnoredIssuingPrincipalChecker:
    def __init__(self, principal_id: str, raise_exception: bool = True):
        self.principal_id = principal_id
        self.raise_exception = raise_exception
    async def __call__(self, request: Request):
        data = Webhook.parse_raw(await request.body())
        if meta := getattr(data, "meta", None):
            # do the meta check 
            if self.raise_exception:
                raise IgnoredPrincipal()
```
then in an app (not globally in the framework)

```python
@app.exception_handler(IgnoredPrincipal)
async def ignored_principal_exception_handler(request: Request, exc: IgnoredPrincipal):
    return JSONResponse(
        status_code=200,
        content={
            "message": f"Principal Ignored: {exc.saleor_event_type}.",
            "success": True,
        },
    )

is_principal_ignored = IgnoredIssuingPrincipalChecker("my-saleor-app", raise_exception=true)

async def product_created(
    payload: List[Webhook], saleor_domain=Depends(saleor_domain_header), _is_principal_ignored=Depends(is_principal_ignored)
):
    print("Product created!")
    print(payload)
```

(code written on "a knee" might not work)

File: src/saleor_app/webhook.py [hunk 2]
```diff
@@ -66,6 +67,7 @@ def decorator(func: WebHookHandlerSignature):
                 dependencies=[
                     Depends(verify_saleor_domain),
                     Depends(verify_webhook_signature),
+                    Depends(verify_issuing_principal_type)
```
Comment (pkucmus) [inline src/saleor_app/webhook.py:70]: As stated above, this should not be enabled globally.

File: src/saleor_app/deps.py (hunk 1 above)
Comment (mstrumeck) [reply src/saleor_app/deps.py]: Done

File: src/saleor_app/webhook.py (hunk 2 above)
Comment (mstrumeck) [reply src/saleor_app/webhook.py]: Done

File: src/saleor_app/schemas/exception_handlers.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from fastapi import Request
+
+from saleor_app.saleor.exceptions import IgnoredPrincipal
+
+
+class IgnoredIssuingPrincipalChecker:
+    def __init__(self, principal_id: str, raise_exception: bool = True):
```
Comment (pkucmus) [inline src/saleor_app/schemas/exception_handlers.py:7]: Now that I think about it, it should be `principal_ids: List[str]`, right?

File: src/saleor_app/schemas/exception_handlers.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        self.principal_id = principal_id
+        self.raise_exception = raise_exception
+
+    async def __call__(self, request: Request):
+        json_data = await request.json()
+        for payload in json_data:
+            if meta := payload.get("meta"):
+                if meta["issuing_principal"]["type"] == self.principal_id:
```
Comment (pkucmus) [inline src/saleor_app/schemas/exception_handlers.py:15]: You are not ignoring an issuer type, you are ignoring a specific issuer so you need to check for `meta["issuing_principal"]["id"]` here

**Review by pkucmus — CHANGES_REQUESTED** [review]

File: src/saleor_app/schemas/exception_handlers.py (hunk 3 above)
Comment (mstrumeck) [reply src/saleor_app/schemas/exception_handlers.py]: Done

File: src/saleor_app/schemas/exception_handlers.py (hunk 4 above)
Comment (mstrumeck) [reply src/saleor_app/schemas/exception_handlers.py]: Done

File: src/saleor_app/tests/test_exception_handlers.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+):
+    ignored_issuing_app_principal = IgnoredIssuingPrincipalChecker(["VXNlcjox"])
+    with pytest.raises(IgnoredPrincipal):
+        await ignored_issuing_app_principal(mock_request_with_metadata)
+
+
+async def test_ignored_issuing_principal_checker_without_raise_exception(
+    mock_request_with_metadata,
+):
+    ignored_issuing_app_principal = IgnoredIssuingPrincipalChecker(["dummy-id"])
```
Comment (kuchichan) [inline src/saleor_app/tests/test_exception_handlers.py:18]: Can we add some assertions, that AsyncMock is actually called with given json input? What is more - if this tests is `without_raise_exception` why we need to use try/except clause?

**Review by pkucmus — APPROVED** [review]

**Pull request merged**